tests/unit/s2n_server_hello_retry_test.c

```diff
@@ -458,6 +623,14 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_io_pair_close(&io_pair));
     }
 
+    /* Hello Retry Request + (poll and no-poll) client hello callback */
+    {
+        /* enable polling */
+        EXPECT_OK(hello_retry_client_hello_cb_test(true));
+        /* disable polling */
+        EXPECT_OK(hello_retry_client_hello_cb_test(false));
```

Comment:
Did you confirm that this test passes before your change? The callback was previously only expected to trigger for the first client hello, not the second / retry?